### DISCUSSION TITLE
ACP-30: Integrate Avalanche Warp Messaging into the EVM

### DISCUSSION THREAD

**Discussion by avalanche-foundation-admin** [avalanche-foundation/ACPs#33] [Discussions]
Integrate Avalanche Warp Messaging into the C-Chain and Subnet-EVM in order to bring Cross-Subnet Communication to the EVM on Avalanche.

https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/30-avalanche-warp-x-evm/README.md

**Comment by patrick-ogrady** [comment]
How would you use AWM in the EVM?
-> Explanation: https://x.com/Michael_Kaplan1/status/1696594547734622382
-> Code: https://github.com/ava-labs/awm-relayer + https://github.com/ava-labs/teleporter
-> E2E Example: https://ohmywarp.com

Shoutout to @michaelkaplan13 for writing these up!
